src/bindings/python/src/pyopenvino/core/common.cpp

```diff
@@ -88,6 +90,109 @@ ov::Tensor tensor_from_numpy(py::array& array, bool shared_memory) {
     return tensor;
 }
 
+ov::PartialShape partial_shape_from_list(const py::list& shape) {
+    using value_type = ov::Dimension::value_type;
+    ov::PartialShape pshape;
+    for (py::handle dim : shape) {
+        if (py::isinstance<py::int_>(dim)) {
+            pshape.insert(pshape.end(), ov::Dimension(dim.cast<value_type>()));
+        } else if (py::isinstance<py::str>(dim)) {
+            pshape.insert(pshape.end(), Common::dimension_from_str(dim.cast<std::string>()));
+        } else if (py::isinstance<ov::Dimension>(dim)) {
+            pshape.insert(pshape.end(), dim.cast<ov::Dimension>());
+        } else if (py::isinstance<py::list>(dim) || py::isinstance<py::tuple>(dim)) {
+            py::list bounded_dim = dim.cast<py::list>();
+            if (bounded_dim.size() != 2) {
+                throw py::type_error("Two elements are expected in tuple(lower, upper) for dynamic dimension, but " +
+                                     std::to_string(bounded_dim.size()) + " were given.");
+            }
+            if (!(py::isinstance<py::int_>(bounded_dim[0]) && py::isinstance<py::int_>(bounded_dim[1]))) {
+                std::string wrong_type = py::isinstance<py::int_>(bounded_dim[0]) ? bounded_dim[1].get_type().str()
+                                                                                  : bounded_dim[0].get_type().str();
+                throw py::type_error("Incorrect type " + wrong_type + " for dynamic dimension, int is expected.");
+            }
+            pshape.insert(pshape.end(),
+                          ov::Dimension(bounded_dim[0].cast<value_type>(), bounded_dim[1].cast<value_type>()));
+        } else {
+            throw py::type_error("Incorrect type " + std::string(dim.get_type().str()) +
+                                 " for dimension. Next types are expected: "
+                                 "int, str, openvino.runtime.Dimension, list or tuple with lower and upper values for "
+                                 "dynamic dimension.");
+        }
+    }
+    return pshape;
+}
+
+bool check_all_digits(const std::string& value) {
+    auto val = ov::util::trim(value);
+    for (const auto& c : val) {
+        if (!std::isdigit(c) || c == '-')
+            return false;
+    }
+    return true;
+}
+
+template <class T>
+T stringToType(const std::string& valStr) {
+    T ret{0};
+    std::istringstream ss(valStr);
+    if (!ss.eof()) {
+        ss >> ret;
+    }
+    return ret;
+}
+
+ov::Dimension dimension_from_str(const std::string& value) {
+    using value_type = ov::Dimension::value_type;
+    auto val = ov::util::trim(value);
+    if (val == "?" || val == "-1") {
+        return {-1};
+    }
+    if (val.find("..") == std::string::npos) {
+        if (!Common::check_all_digits(val))
+            IE_THROW() << "Cannot parse dimension: \"" << val << "\"";
```

Comment:
Is `IE_TROW()` still desired? I think you should use `OPENVINO_ASSERT(...)` instead. Applies to the rest of the code as well.

Reply:
done